icu4c/source/i18n/usearch.cpp

```diff
@@ -2313,7 +2387,7 @@ U_CAPI UBool U_EXPORT2 usearch_searchBackwards(UStringSearch  *strsrch,
 UBool usearch_handleNextExact(UStringSearch *strsrch, UErrorCode *status)
 {
     if (U_FAILURE(*status)) {
-        setMatchNotFound(strsrch);
+        setMatchNotFound(strsrch, *status);
```

Comment:
`setMatchNotFound()` has been changed to take no action when there is an incoming error.
Does it need to reset the matching state to no match even when there is an error? As is guaranteed to be the case here.

Reply:
We should probably just delete this line. If this function is called with an incoming error, then it should just return without doing anything, as usual.

Reply:
Hm, that's a good point Andy -- thank you.

Before, the `setMatchNotFound` function would have *always* set the state to "no match", but now it will do nothing and simply exit if there is an incoming error. The API docs for `usearch_next`/`usearch_previous` don't specifically call out what happens if an error occurs, but I think we should probably retain the previous behavior, so that the state will be set to "no match" on error.

I was trying to think of ways to add a test case for this, but there doesn't really seem to be any good way to test this. The `setMatchNotFound` is called at the beginning of `usearch_handle*` if there is an error, but these methods are not exported, so the test code can't directly call them. So we'd need to have `usearch_next`/`usearch_previous` somehow fail in the middle of execution, since they also check for failure first before doing anything.
I don't know if we'd want to export the `usearch_handle*` functions just for testing though (?).

I'll leave this w/o testing for now, and update the `setMatchNotFound` to always set "no match" as before.

Reply:
I still prefer to handle errors consistently -- and bail out early if there is an error.
But I don't feel strongly enough to overrule either of you.
It just seems like every time we make an exception, we will surprise ourselves, and that adds to the maintenance burden.

Reply:
Filed a follow-up ticket for ICU 71 here: https://unicode-org.atlassian.net/browse/ICU-21761